cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -220,7 +261,8 @@ Injector createPreviewInjector(ApplicationId applicationId) throws IOException {
       .filter(s -> s.endsWith(".bind.address"))
       .forEach(key -> previewCConf.set(key, localhost));
 
-    Path previewDir = Files.createDirectories(previewDataDir.resolve(applicationId.getApplication()));
+    Path previewDir = Files.createDirectories(previewDataDir.resolve(
+      programId.toString().replaceFirst(PROGRAM_PREFIX, "")));
```

Comment:
Same here. It's better to construct the path explicitly rather than using the toString format and assuming it has some prefix.